packages/realm-app-importer/src/AppImporter.ts

```diff
@@ -509,63 +345,17 @@ export class AppImporter {
 
         const currentProvider = providers.find((provider) => provider.type === config.type);
         if (currentProvider) {
-          const url = `${this.apiUrl}/groups/${groupId}/apps/${appId}/auth_providers/${currentProvider._id}`;
-          const response = await fetch(url, {
-            method: "PATCH",
-            headers: {
-              Authorization: `Bearer ${this.accessToken}`,
-              "content-type": "application/json",
-            },
-            body: JSON.stringify({ ...config, _id: currentProvider._id }),
-          });
-          if (!response.ok) {
-            const json = await response.json();
-            const error = isErrorResponse(json) ? json.error : "No error message";
-            console.error("Failed to apply auth_provider: ", { config, url, error });
-          }
+          await this.client.patchAuthProvider(appId, currentProvider._id, config);
         } else {
-          const url = `${this.apiUrl}/groups/${groupId}/apps/${appId}/auth_providers`;
-          const response = await fetch(url, {
-            method: "POST",
-            headers: {
-              Authorization: `Bearer ${this.accessToken}`,
-              "content-type": "application/json",
-            },
-            body: JSON.stringify(config),
-          });
-          if (!response.ok) {
-            const json = await response.json();
-            const error = isErrorResponse(json) ? json.error : "No error message";
-            console.error("Failed to apply auth_provider: ", { config, url, error });
-          }
+          await this.client.createAuthProvider(appId, config);
         }
       }
+      // Wait a bit for BaaS to enable the auth provider
+      await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
This is really annoying and something that I'd like to get rid of before merging.